src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/mappers/CancellationReasonMapper.kt

```diff
@@ -0,0 +1,18 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.mappers
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.CancellationReason
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.CancellationReasonRepository
+
+@Component
+class CancellationReasonMapper(
+  private val cancellationReasonRepository: CancellationReasonRepository
+) {
+  fun mapCancellationReasonIdToCancellationReason(cancellationReasonId: String): CancellationReason {
+    return cancellationReasonRepository.findByIdOrNull(cancellationReasonId)
+      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "Cancellation reason not found [id=$cancellationReasonId]")
```

Comment:
i'm not sure 404 is the appropriate response here. if we raised `EntityNotFound` we would also return 404. but i think bad request might be better. i.e. the cancellation code is invalid

Reply:
or have this method raise entitynotfound, and have the controller catch it and return bad request